main.js

```diff
@@ -160,4 +160,146 @@ const inventory = [
     originalStock: 10,
     sold: 8,
   },
-];
\ No newline at end of file
+];
+
+// Opdracht 1a: Hoeveel exemplaren moeten we nog verkopen? Schrijf een functie die dit berekent.
+// aantal nog te verkopen units is gelijk aan (originalStock - sold) per type en dan totaal berekenen.
+const typesToSell = inventory.map((unit) => {
+  return unit.originalStock - unit.sold
+})
+console.log(typesToSell)
+
+const totalUnitsToSell = typesToSell.reduce((accumulator, currentValue) => {
+  return accumulator + currentValue;
+})
+console.log(totalUnitsToSell);
+
+
+//Opdracht 1b: Zorg ervoor dat dit aantal in het rood wordt weergegeven op de pagina => css
+document.getElementById("input1").value = totalUnitsToSell;
+
+//Opdracht 2a: Gebruik een array-methode om een array te maken met alle tv-type namen.
+tvTypes = inventory.map((typeName) => {
+  return typeName.type;
+})
+console.log(tvTypes)
+
+// Opdracht 2b: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die volledig uitverkocht zijn.
+tvSoldOut = inventory.filter((count) => {
+  if (count.originalStock - count.sold === 0){
+    return count;
+  }
+})
+console.log(tvSoldOut);
+
+// Opdracht 2c: Gebruik een array-methode om alle tv's te verzamelen (de hele objecten) die over AmbiLight beschikken.
+tvAmbilight = inventory.filter((light) => {
+  if (light.options.ambiLight){
+    return light;
+  }
+})
+console.log(tvAmbilight);
+
+// Opdracht 2d: Schrijf een functie die alle tv's van laagste naar hoogste prijs sorteert.
+tvPriceSorted = inventory.sort((previousPrice, currentPrice) => {
+  return previousPrice.price - currentPrice.price ;
+})
+console.log(tvPriceSorted);
+
+//Opdracht 3a: Wat is onze doel-opbrengst? Bereken wat de totale opbrengst is, als we alle exemplaren van ieder type
+// zouden verkopen.
+profitPerType = inventory.map((unit) => {
+  return (unit.originalStock - unit.sold)* unit.price;
```

Comment:
Kleine vergissing. De bedoeling is dat we berekenen wat de totale opbrengst zou zijn als alles is verkocht. Ongeacht wat er al verkocht is. Dus alle original stock * price = 26727